druid/examples/calc.rs

```diff
@@ -112,95 +115,239 @@ impl CalcState {
+fn digit_button(digit: u8) -> Padding<CalcState> {
     pad(Button::new(
         format!("{}", digit),
         move |_ctx, data: &mut CalcState, _env| data.digit(digit),
     ))
 }
 
-fn flex_row<T: Data>(
-    w1: impl Widget<T> + 'static,
-    w2: impl Widget<T> + 'static,
-    w3: impl Widget<T> + 'static,
-    w4: impl Widget<T> + 'static,
-) -> impl Widget<T> {
-    let mut row = Row::new();
-    row.add_child(w1, 1.0);
-    row.add_child(w2, 1.0);
-    row.add_child(w3, 1.0);
-    row.add_child(w4, 1.0);
-    row
-}
+struct ButtonLabel(ButtonId, Option<&'static str>);
 
 fn build_calc() -> impl Widget<CalcState> {
+    use crate::ButtonId::*;
+
     let mut column = Column::new();
     let display = LensWrap::new(
         DynLabel::new(|data: &String, _env| data.clone()),
         lenses::calc_state::value,
     );
+
+    let button_rows = [
+        [
+            ButtonLabel(Op('c'), Some("CE")),
+            ButtonLabel(Op('C'), None),
+            ButtonLabel(Op('⌫'), None),
+            ButtonLabel(Op('÷'), None),
+        ],
+        [
+            ButtonLabel(Digit(7), None),
+            ButtonLabel(Digit(8), None),
+            ButtonLabel(Digit(9), None),
+            ButtonLabel(Op('×'), None),
+        ],
+        [
+            ButtonLabel(Digit(4), None),
+            ButtonLabel(Digit(5), None),
+            ButtonLabel(Digit(6), None),
+            ButtonLabel(Op('−'), None),
+        ],
+        [
+            ButtonLabel(Digit(1), None),
+            ButtonLabel(Digit(2), None),
+            ButtonLabel(Digit(3), None),
+            ButtonLabel(Op('+'), None),
+        ],
+        [
+            ButtonLabel(Digit(0), None),
+            ButtonLabel(Op('±'), None),
+            ButtonLabel(Op('.'), None),
+            ButtonLabel(Op('='), None),
+        ],
+    ];
+
     column.add_child(pad(display), 0.0);
-    column.add_child(
-        flex_row(
-            op_button_label('c', "CE".to_string()),
-            op_button('C'),
-            op_button('⌫'),
-            op_button('÷'),
-        ),
-        1.0,
-    );
-    column.add_child(
-        flex_row(
-            digit_button(7),
-            digit_button(8),
-            digit_button(9),
-            op_button('×'),
-        ),
-        1.0,
-    );
-    column.add_child(
-        flex_row(
-            digit_button(4),
-            digit_button(5),
-            digit_button(6),
-            op_button('−'),
-        ),
-        1.0,
-    );
-    column.add_child(
-        flex_row(
-            digit_button(1),
-            digit_button(2),
-            digit_button(3),
-            op_button('+'),
-        ),
-        1.0,
-    );
-    column.add_child(
-        flex_row(
-            op_button('±'),
-            digit_button(0),
-            op_button('.'),
-            op_button('='),
-        ),
-        1.0,
-    );
-    column
+    let mut key_map = HashMap::new();
+    for (col_id, buttons) in button_rows.iter().enumerate() {
```

Comment:
to check my understanding: this key map isn't actually used, it's just sort of showing how we _could_ address buttons?

Reply:
Yes, exactly